### DISCUSSION TITLE
Tolerance in lag separation

### DISCUSSION THREAD

**Discussion by rlsandovalp** [GeoStat-Framework/GSTools#307] [Ideas]
Hi!

As far as I understand, the vario_estimate function allows you to define the bin edges. Then, in a position associated with the center of each bin, the function calculates the experimental variogram by using all the points separated by a distance between the min value of the bin and the max value of the bin. 

I wonder if there is a way of using the same function but specifying uniform spacing (separation) and tolerance (tol) like in the following image. In such a case, the variogram of the first position would be computed by employing all the points separated by a distance equal to the specified separation +- the specified tolerance.
![image](https://user-images.githubusercontent.com/51087450/235982727-d51b6c3b-f96f-4f8f-a6d4-a5d142a00df5.png)

I ask because we came up with a method to manipulate the results of vario_estimate and recover this type of info for tolerance values equal to or smaller than half the separation (first two cases of the next figure). However, we couldn't do it for tolerance values larger than half the separation (third case of the next figure) which in some cases can also be useful.

![image](https://user-images.githubusercontent.com/51087450/235985319-56990d99-d81b-49f1-a101-8c74f9ddb97b.png)

Leonardo

**Comment by MuellerSeb** [comment]
Hey Leonardo,

this could be a simple extension, since we only need to modify two lines when the point-pair distance is compared to the bin edges.

We will consider this!

Cheers,
Sebastian
